src/crawlee/service_container.py

```diff
@@ -18,50 +16,51 @@
     'get_storage_client',
     'set_local_storage_client',
     'set_cloud_storage_client',
-    'set_default_storage_client_type',
     'get_configuration',
-    'get_configuration_if_set',
     'set_configuration',
     'get_event_manager',
     'set_event_manager',
+    'StorageClientType',
 ]
 
 StorageClientType = Literal['cloud', 'local']
 
 
-class _Services(TypedDict):
-    local_storage_client: NotRequired[BaseStorageClient]
-    cloud_storage_client: NotRequired[BaseStorageClient]
-    configuration: NotRequired[Configuration]
-    event_manager: NotRequired[EventManager]
+@dataclass
+class _Services:
```

Comment:
It might be useful to pull the type out so that we can require it in places where we may want to override the default service container - e.g., `BasicCrawlerOptions`.

Reply:
We do not need it in the end as the services can be still passed directly to the components.